configuration/postgres/postgres.go

```diff
@@ -162,42 +163,40 @@ func (p *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequ
 }
 
 func (p *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
-	var pgNotifyChannels []string
+	pgNotifyChannel := ""
 	for k, v := range req.Metadata {
-		if res := strings.EqualFold(pgNotifyChannelKey, k) && !slices.Contains(pgNotifyChannels, v); res {
-			pgNotifyChannels = append(pgNotifyChannels, v) // append unique channel names only
+		if strings.EqualFold(pgNotifyChannelKey, k) {
+			pgNotifyChannel = v
 		}
 	}
-	if len(pgNotifyChannels) == 0 {
+	if pgNotifyChannel == "" {
 		return "", fmt.Errorf("unable to subscribe to '%s'.pgNotifyChannel attribute cannot be empty", p.metadata.ConfigTable)
 	}
-	return p.subscribeToChannel(ctx, pgNotifyChannels, req, handler)
+	return p.subscribeToChannel(ctx, pgNotifyChannel, req, handler)
 }
 
 func (p *ConfigurationStore) Unsubscribe(ctx context.Context, req *configuration.UnsubscribeRequest) error {
 	p.configLock.Lock()
 	defer p.configLock.Unlock()
-	for k, v := range p.ActiveSubscriptions {
-		if v.uuid == req.ID {
-			if oldStopChan, ok := p.subscribeStopChanMap[req.ID]; ok {
-				delete(p.subscribeStopChanMap, req.ID)
-				close(oldStopChan)
-				pgChannel := fmt.Sprintf(unlistenTemplate, k)
-				conn, err := p.client.Acquire(ctx)
-				if err != nil {
-					p.logger.Errorf("error acquiring connection:", err)
-					return fmt.Errorf("error acquiring connection: %w ", err)
-				}
-				defer conn.Release()
-				_, err = conn.Exec(ctx, pgChannel)
-				if err != nil {
-					p.logger.Errorf("error un-listening to channel:", err)
-					return fmt.Errorf("error un-listening to channel: %w", err)
-				}
-				delete(p.ActiveSubscriptions, k)
-				return nil
-			}
+	if sub, ok := p.ActiveSubscriptions[req.ID]; ok {
+		if oldStopChan, ok := p.subscribeStopChanMap[req.ID]; ok {
+			delete(p.subscribeStopChanMap, req.ID)
+			close(oldStopChan)
 		}
+		pgChannel := fmt.Sprintf(unlistenTemplate, sub.channel)
```

Comment:
Since the constants `unlistenTemplate` and `listenTemplate` are only used once, please inline them. It makes the code a lot more readable...

(Also consider using simple string concatenation: `pgChannel := "UNLISTEN " + sub.channel` )